packages/forms/src/UIForm/UIForm.container.js

```diff
@@ -51,6 +52,24 @@ export default class UIForm extends React.Component {
 	}
 
 	/**
+	 * Reset the forms data and errors. Triggers the onReset callback
+	 * @param event The reset event
+	 */
+	onReset(event) {
+		this.updateForm(
+			this.props.formName,
+			this.props.data.jsonSchema,
+			this.props.data.uiSchema,
+			this.props.data.properties
+		);
+		this.setErrors(this.props.formName, {});
+
+		if (this.props.onReset) {
+			this.props.onReset(event);
+		}
+	}
```

Comment:
In fact it was because of the initial values that were not available in the component but only in the container/connector's props.
But you're right, I'll change that and passe the original data to the component